File: library/src/main/java/me/proxer/library/api/ucp/UcpTopTenEndpoint.java
package me.proxer.library.api.ucp;

import lombok.experimental.Accessors;
import me.proxer.library.api.Endpoint;
import me.proxer.library.api.ProxerCall;
import me.proxer.library.entity.ucp.UcpTopTenEntry;

import java.util.List;

/**
 * Endpoint for unmarking an entry as favorite.
 *
 * @author Ruben Gees
 */
@Accessors(fluent = true)
public final class UcpTopTenEndpoint implements Endpoint<List<UcpTopTenEntry>> {

    private final InternalApi internalApi;

    UcpTopTenEndpoint(final InternalApi internalApi) {
        this.internalApi = internalApi;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public ProxerCall<List<UcpTopTenEntry>> build() {
        return internalApi.topTen();
    }
}

File: library/src/main/kotlin/me/proxer/library/api/ucp/UcpTopTenEndpoint.kt
package me.proxer.library.api.ucp

import me.proxer.library.ProxerCall
import me.proxer.library.api.Endpoint
import me.proxer.library.entity.ucp.UcpTopTenEntry

/**
 * Endpoint for requesting the top ten list of the current user.
 *
 * @author Ruben Gees
 */
class UcpTopTenEndpoint internal constructor(
    private val internalApi: InternalApi
) : Endpoint<List<UcpTopTenEntry>> {

    override fun build(): ProxerCall<List<UcpTopTenEntry>> {
        return internalApi.topTen()
    }
}
